### ISSUE TITLE
[Bug] Updating lightweight monitor in project generates error

### ISSUE THREAD

**Issue by renzedj** [elastic/synthetics#843]
# Issue

Attempting to update a lightweight monitor in a project generates the following error:

```
> security_exception
 Root causes:
   security_exception: action [indices:data/read/search] is unauthorized for API key id [<key-id>] of user [<my-user-id>], this action is granted by the index privileges [read,all]
     Failed to update journey: 7f5243e1-b593-4639-ba42-f081c802c79d
```

> **Note:** The user ID shows my personal ID (as expected), however I am the Elastic Cloud admin for our organization.

# Steps to Replicate Error

1. Create the following lightweight test in an `@elastic/synthetics` project:
```yaml
heartbeat.monitors:
  - id: '7f5243e1-b593-4639-ba42-f081c802c79d'
    name: 'Google'
    type: 'http'
    enabled: true
    alert.status.enabled: true
    check.request.method: 'GET'
    response:
      include_body: 'on_error'
      include_headers: true
    urls: 'https://google.com/'
    schedule: '@every 5m'
    timeout: '15s'
```
2. Push to Kibana: `SYNTHETICS_API_KEY=<api_key> npm run push`
3. Change `alert.status.enabled: false` 
4. Push to Kibana: `SYNTHETICS_API_KEY=<api_key> npm run push`

> **Notes:**
> * I have generated a new API key in the current version of Kibana (8.10.x) to validate that this isn't an issue that's already been resolved in Kibana.
> * Workaround is to update the ID, which deletes the old monitor and creates the new, but this doesn't preserve monitor history.

# Versions

* Elastic Cloud: 8.10.0, 8.10.2
* `@elastic/synthetics`: 1.4.0

**Comment by vigneshshanmugam** [comment]
This bug applies to all browser monitors as well. Seems like a Kibana API issue @shahzad31 @awahab07 FYI.

**Comment by renzedj** [comment]
> This bug applies to all browser monitors as well. Seems like a Kibana API issue @shahzad31 @awahab07 FYI.

You're referring to the scripted browser monitors, yes? I haven't encountered it with them, but then, I haven't updated any of them in awhile. I first noticed this when I upgraded to 1.4.0, if that helps any, but again, I hadn't been working in Synthetics for a few months before that.

**Comment by shahzad31** [comment]
So I have been unable to reproduce this. Yeah it's just standard api key but also has permissions to kibana features. You can take a look in api key management in kibana stack management.

So the user which you used to generate this api key. Is it standard user?
And regenerating api key doesn't solves the issue?

**Comment by renzedj** [comment]
I generated it as myself, and I'm assigned to the `admin` role through SSO. I haven't tried logging in as the default admin user yet to do this. As far as checking API key management under Stack Management, I don't see these keys listed there. I see a lot of keys there for alerts, but none for synthetics. I'll look again though.

**Comment by renzedj** [comment]
As an update: I just generated a key by logging into our Elastic Cloud instance with the default admin user through cloud.elastic.co and received the same error. Could this have something to do with the fact that I upgraded from an earlier version of `@elastic/synthetics` (v1.2.0, I believe) to v1.4.0?

**Comment by shahzad31** [comment]
@renzedj keys privileges looks fine to me.  i am trying to dig deep. Things is that updating monitor shouldn't even do that indices search call for which it's throwing error. Perhaps you can create an API key with giving permissions for cluster read against indices. 

i will keep you posted when i can reproduce and have workaround api key priviliges

**Comment by renzedj** [comment]
@shahzad31 -

Based on the [documentation](https://www.elastic.co/guide/en/observability/current/synthetics-role-write.html), the `cluster` permissions do not appear to be necessary. I tried the following, and it works as expected:

```json
{
  "name": "synthetics-test-key-001`",
  "expiration": "1d",
  "role_descriptors": {
    "synthetics_writer": {
      "cluster": [],
      "indices": [
        {
          "names": [
            "synthetics-*"
          ],
          "privileges": [
            "read"
          ]
        }
      ],
      "applications": [
        {
          "application": "kibana-.kibana",
          "privileges": [
            "feature_uptime.all"
          ],
          "resources": [
            "*"
          ]
        }
      ],
      "run_as": [],
      "metadata": {},
      "transient_metadata": {
        "enabled": true
      }
    }
  }
}
```

**Comment by shahzad31** [comment]
Yeah that's really weird. i need to check why it's making call to read indices.

**Comment by renzedj** [comment]
Thx. I was going to try to figure out the permissions today anyway, you really helped shortcut that process. Glad the workaround is so easy.

**Comment by shahzad31** [comment]
@renzedj about your this comment 

`Workaround is to update the ID, which deletes the old monitor and creates the new, but this doesn't preserve monitor history.`
if you keep the monitor.id same in your config in project monitor, deleting and recreating should preserve your history in app. 
Since we don't delete the associated data when we delete monitor.

**Comment by renzedj** [comment]
@shahzad31 - Thanks, that was a quick turnaround!

**Comment by shahzad31** [comment]
> @shahzad31 - Thanks, that was a quick turnaround!

You are welcome, so this will be shipping with 8.10.3 patch, which should be released pretty soon and the upcoming 8.11.0
